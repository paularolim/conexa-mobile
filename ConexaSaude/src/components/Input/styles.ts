import { TextInput, View } from 'react-native';
import styled from 'styled-components/native';

import { Text } from '@components/Text';

export const Container = styled(View)`
  background-color: ${({ theme }) => theme.colors.white};
  width: 100%;
  justify-content: center;
  align-items: flex-start;
`;

export const Label = styled(Text)`
  color: ${({ theme }) => theme.colors.gray};
  font-size: 18px;
`;

export const FieldContainer = styled(View)`
  border: 1px solid ${({ theme }) => theme.colors.gray};
  border-radius: 4px;
  width: 100%;
  flex-direction: row;
  align-items: center;
  min-height: 48px;
`;

export const Field = styled(TextInput)`
  color: ${({ theme }) => theme.colors.black};
  flex-grow: 1;
  padding: 8px 16px;
  flex: 1;
`;

export const IconContainer = styled(View)`
  padding-right: 16px;
`;
