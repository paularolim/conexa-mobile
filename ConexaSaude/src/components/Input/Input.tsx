import { useState } from 'react';
import { TouchableOpacity } from 'react-native';
import { useTheme } from 'styled-components/native';

import { Icon } from '@components/Icon';

import { Container, Field, FieldContainer, Label } from './styles';
import { InputProps } from './types';

export function Input({ label, hasPasswordIcon, secureTextEntry, ...rest }: InputProps) {
  const [hidePassword, setHidePassword] = useState(secureTextEntry);

  const { colors } = useTheme();

  const togglePasswordVisibility = () => {
    setHidePassword((prev) => !prev);
  };

  return (
    <Container>
      <Label>{label}</Label>
      <FieldContainer>
        <Field testID="field" secureTextEntry={hidePassword} {...rest} />
        {hasPasswordIcon && (
          <TouchableOpacity onPress={togglePasswordVisibility} testID="password-visibility-toggle">
            <Icon
              name={hidePassword ? 'eye-on' : 'eye-off'}
              width={25}
              height={25}
              fill={colors.gray}
            />
          </TouchableOpacity>
        )}
      </FieldContainer>
    </Container>
  );
}
