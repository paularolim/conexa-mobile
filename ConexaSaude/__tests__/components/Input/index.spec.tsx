import { render, fireEvent } from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import { Input } from '@components/Input';

import { Wrapper } from '../../Wrapper';

describe('Input component', () => {
  it('should render component correctly', () => {
    const tree = renderer
      .create(
        <Wrapper>
          <Input label="Test Input" />
        </Wrapper>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should change password visibility on press icon', () => {
    const { getByTestId } = render(
      <Wrapper>
        <Input label="Test Input" secureTextEntry hasPasswordIcon />
      </Wrapper>,
    );

    const field = getByTestId('field');
    const toggle = getByTestId('password-visibility-toggle');

    fireEvent.press(toggle);
    expect(field.props.secureTextEntry).toBe(false);

    fireEvent.press(toggle);
    expect(field.props.secureTextEntry).toBe(true);
  });
});
