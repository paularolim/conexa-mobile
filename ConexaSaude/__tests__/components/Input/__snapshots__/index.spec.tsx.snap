// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component should render component correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "flex-start",
        "backgroundColor": "#FFFFFF",
        "justifyContent": "center",
        "width": "100%",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#000000",
        },
        {
          "color": "#707683",
          "fontSize": 18,
        },
      ]
    }
  >
    Test Input
  </Text>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderColor": "#707683",
          "borderRadius": 4,
          "borderStyle": "solid",
          "borderWidth": 1,
          "flexDirection": "row",
          "minHeight": 48,
          "width": "100%",
        },
      ]
    }
  >
    <TextInput
      style={
        [
          {
            "color": "#000000",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "paddingBottom": 8,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 8,
          },
        ]
      }
      testID="field"
    />
  </View>
</View>
`;
